lib/utils.js

```diff
@@ -219,16 +219,67 @@ utils.findRefs = (json, absolutePath, relativePath, initialFormat, asyncapiYAMLo
 utils.getLocationOf = (jsonPointer, asyncapiYAMLorJSON, initialFormat) => {
   const ast = getAST(asyncapiYAMLorJSON, initialFormat);
   if (!ast) return { jsonPointer };
+
   return findLocationOf(jsonPointerToArray(jsonPointer), ast, initialFormat);
 };
 
 utils.improveAjvErrors = (errors, asyncapiYAMLorJSON, initialFormat) => {
   const ast = getAST(asyncapiYAMLorJSON, initialFormat);
   return errors.map(error => {
     const defaultLocation = { jsonPointer: error.dataPath || '/' };
+    console.log(error.dataPath);
     return {
       title: `${error.dataPath || '/'} ${error.message}`,
       location: ast ? findLocationOf(jsonPointerToArray(error.dataPath), ast, initialFormat) : defaultLocation,
     };
   });
-};
\ No newline at end of file
+};
+
+/**
+ * Returns an array with all strings in between curly braces, including braces
+ */
+utils.getCurlyBracesEl = str => {
+  if (typeof str !== 'string') return;
+  const regex = new RE2(/{(.+?)}/g);
+
+  return regex.match(str); 
+};
+
+/**
+ * Returns an array with values that map to the keys of provided object
+ */
+utils.groupExistingElements = (arr, obj) => {
```

Comment:
Same as above. It's not clear what's the purpose of this function. Also, it seems you named this function when you thought it was going to be only an array of existing elements but now it also includes `noEl`, which I don't understand what it is.

Reply:
I hate reading my code a week after writing it and not understanding my intentions...
anyway 😅  fixed the code that was overcomplicated